src/core_plugins/metrics/public/components/vis_editor.js

```diff
@@ -108,9 +150,26 @@ class VisEditor extends Component {
               model={model}
               visData={this.props.visData}
               dateFormat={this.props.config.get('dateFormat')}
-              onChange={handleChange}
+              onChange={this.handleChange}
             />
           </div>
+          <ModalConfirm
+            show={this.state.showConfirm}
+            onConfirm={this.handleConfirm}
+            onCancel={this.handleCancel}
+            title="Incompatible Metrics"
+            confirmButtonText="Yes, remove metrics"
+            message={
+              <div>
+                <p>
+                  The <strong>data timerange mode</strong> you have chosen is not
+                  compatible with some of the metrics you have configured. If you
+                  proceed the incompatible metrics will be removed.
+                </p>
+                <p>Are you sure you want to do this?</p>
```

Comment:
Avoiding `Are you sure?` is recommended by the EUI writing guide (see earlier comment).